src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -830,27 +833,35 @@ public boolean accept(File file)
 
     private List<File> listSnapshots()
     {
-        final List<File> snapshots = new LinkedList<>();
-        for (final File dir : dataPaths)
+        SNAPSHOT_LOCK.readLock().lock();
```

Comment:
Does this have a timeout, or is forever blocking? If not, may make sense to fail after >60 seconds, so we are aware of a deadlock scenario?